### PR TITLE
[Merged by Bors] - chore(LinearAlgebra/Matrix/Determinant/TotallyUnimodular): golf proof

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by madvorak — APPROVED** [review]

File: Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean [hunk 1]
```diff
@@ -41,26 +41,17 @@ lemma isTotallyUnimodular_iff (A : Matrix m n R) : A.IsTotallyUnimodular ↔
...
-      else
-        left
-        unfold Function.Injective at hg
-        push_neg at hg
-        obtain ⟨i, j, hqij, hij⟩ := hg
-        apply Matrix.det_zero_of_column_eq hij
-        intro
-        simp [hqij]
+    if h : f.Injective ∧ g.Injective then
+        exact hA k f g h.1 h.2
```
Comment (madvorak) [inline Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean:45]: one indentation less IMO

File: Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean [hunk 2]
```diff
@@ -41,26 +41,17 @@ lemma isTotallyUnimodular_iff (A : Matrix m n R) : A.IsTotallyUnimodular ↔
...
       left
-      unfold Function.Injective at hf
-      push_neg at hf
-      obtain ⟨i, j, hpij, hij⟩ := hf
-      apply Matrix.det_zero_of_row_eq
-      · exact hij
-      show (A (f i)) ∘ (g ·) = (A (f j)) ∘ (g ·)
-      rw [hpij]
+      simp only [Function.Injective, not_and_or, not_forall, ← ne_eq] at h
+      obtain ⟨i, j, hqij, hij⟩ | ⟨i, j, hqij, hij⟩ := h
```
Comment (madvorak) [inline Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean:49]: When we are at it `hqij` should be `hfij` everywhere.

File: Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean [hunk 3]
```diff
@@ -41,26 +41,17 @@ lemma isTotallyUnimodular_iff (A : Matrix m n R) : A.IsTotallyUnimodular ↔
...
     else
       left
-      unfold Function.Injective at hf
-      push_neg at hf
-      obtain ⟨i, j, hpij, hij⟩ := hf
-      apply Matrix.det_zero_of_row_eq
-      · exact hij
-      show (A (f i)) ∘ (g ·) = (A (f j)) ∘ (g ·)
-      rw [hpij]
+      simp only [Function.Injective, not_and_or, not_forall, ← ne_eq] at h
```
Comment (Ruben-VandeVelde) [inline Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean:48]:
```suggestion
      simp_rw [not_and_or, Function.not_injective_iff] at h
```

File: Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean [hunk 4]
```diff
@@ -41,26 +41,17 @@ lemma isTotallyUnimodular_iff (A : Matrix m n R) : A.IsTotallyUnimodular ↔
...
+      obtain ⟨i, j, hfij, hij⟩ | ⟨i, j, hrij, hij⟩ := h
+      · rw [← det_transpose, transpose_submatrix]
+        apply det_zero_of_column_eq hij.symm
+        simp [hfij]
+      · apply det_zero_of_column_eq hij
+        simp [hrij]
```
Comment (Ruben-VandeVelde) [inline Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean:54]:
```suggestion
      obtain ⟨i, j, hfij, hij⟩ | ⟨i, j, hgij, hij⟩ := h
      · rw [← det_transpose, transpose_submatrix]
        apply det_zero_of_column_eq hij.symm
        simp [hfij]
      · apply det_zero_of_column_eq hij
        simp [hgij]
```

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**